src/pke/unittest/utckksrns/UnitTestCKKSrnsCompositeScaling.cpp

```diff
@@ -177,154 +177,185 @@ constexpr usint BATCH         = 8;
+    // TestType,  Descr, Scheme,         RDim,     MultDepth, SModSize, DSize, BatchSz, SecKeyDist,      MaxRelinSkDeg, FModSize, SecLvl,       KSTech, ScalTech,               LDigits, PtMod, StdDev, EvalAddCt, KSCt, MultTech, EncTech, PREMode, MultipartyMode, decryptionNoiseMode, ExecutionMode, NoiseEstimate, RegisterWordSize, CompositeDegree, Slots
+    { ADD_PACKED, "11", {CKKSRNS_SCHEME, RING_DIM, 7,         63,       DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,          64,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 27, 3},   0},
+    { ADD_PACKED, "12", {CKKSRNS_SCHEME, RING_DIM, 7,         90,       DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,          96,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 27, 4},   0},
+    { ADD_PACKED, "13", {CKKSRNS_SCHEME, RING_DIM, 7,         118,      DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,         120,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 64, 2},   0},
+    { ADD_PACKED, "14", {CKKSRNS_SCHEME, RING_DIM, 7,         52,       DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,          54,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 27, 2},   0},
+    { ADD_PACKED, "15", {CKKSRNS_SCHEME, RING_DIM, 7,         89,       DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,          96,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 32, 4},   0},
+    { ADD_PACKED, "16", {CKKSRNS_SCHEME, RING_DIM, 7,         108,      DSIZE, BATCH,   UNIFORM_TERNARY, DFLT,         118,       HEStd_NotSet, HYBRID, COMPOSITESCALINGMANUAL, DFLT,    DFLT,  DFLT,   DFLT,      DFLT, DFLT,     DFLT,    DFLT,    DFLT, DFLT, DFLT, DFLT, 48, 3},   0},
     // ==========================================
-    // TestType,  Descr, Scheme,         RDim, MultDepth, SModSize, DSize, BatchSz, SecKeyDist, MaxRelinSkDeg, FModSize, SecLvl,       KSTech, ScalTech,             LDigits, PtMod, StdDev, EvalAddCt, KSCt, MultTech, EncTech, PREMode, MultipartyMode, decryptionNoiseMode, ExecutionMode, NoiseEstimate, RegisterWordSize, Slots
+    // TestType,  Descr, Scheme,         RDim,      MultDepth, SModSize, DSize, BatchSz, SecKeyDist, MaxRelinSkDeg, FModSize, SecLvl,       KSTech, ScalTech,              LDigits, PtMod, StdDev, EvalAddCt, KSCt, MultTech, EncTech, PREMode, MultipartyMode, decryptionNoiseMode, ExecutionMode, NoiseEstimate, RegisterWordSize, CompositeDegree, Slots
     // composite degree d = 2
```

Comment:
Why are there comments with composite degree values (ex: "// composite degree d = 2")? if the degree is set, then it should be set explicitly. Where is it set?

Reply:
In this case, compositeDegree is calculated automatically since it is set to COMPOSITESCALINGAUTO. This comment is just to indicate, and differentiate, what those parameter values (firstModSize and scalModSize) result in as per compositeDegree value. If not necessary, I can remove the comments.

Reply:
so, if COMPOSITESCALINGAUTO is used then we should not allow users to call SetCompositeDegree(). In this case src/pke/lib/scheme/gen-cryptocontext-params-validation.cpp should be updated.

Reply:
@dsuponitskiy Done. Please see latest commits.